### ISSUE TITLE
[question] Docker conan remote authentication fails

### ISSUE THREAD

**Issue by andreear-23** [conan-io/conan#13153]
### What is your question?

Hi,

I'm trying to build a project that uses conan packages but it requires a specific compiler (gcc 7.3) for using pre-build packages. For that I'm trying to build it inside a docker image (based on centos) that already has the compiler installed. 
However, the packages are stored on a private conan remote for which I have to authenticate. My problem is that authentication fails inside the docker image with:
``` 
ERROR: b'<html>\r\n<head><title>403 Forbidden</title></head>\r\n<body>\r\n<center><h1>403 Forbidden</h1></center>\r\n</body>\r\n</html>\r\n'                             
                                                                                                                                                                         
Invalid server response, check remote URL and try again.
````
I don't know if I'm supposed to setup any specific parameters (port forwarding or anything else) when running the docker image, this is how I run it:
```
docker run --rm -it -p 80:80 -v ~/workspace/git/SAS:/workspace/SAS centos_bld:1.3 /bin/bash
```

I can manage to authenticate to the conan remote on my host machine just fine, the problem is doing it in the docker image. 

Any ideas of what I'm doing wrong? 
Thanks!

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @AndreeaRadoescu 

Is it possible that you used the wrong URL in the docker? it could be that you used the web browser URL instead of the correct API endpoint? If it is your private server, in Artifactory is the "SetMeUp" button URL. Can you please check your remote?

**Comment by andreear-23** [comment]
Hi @memsharded,
I don't think so, I used the same commands for both host and docker.
Basically:
```
conan remote add conan-thirdparty https://<private_domain>.com/artifactory/api/conan/conan-thirdparty
conan user -r conan-thirdparty <user> -p
```
On my host if I write `conan user` I see the authentication is successful:
```
Current user of remote 'conan-thirdparty' set to: '<user>' [Authenticated]
```
And on the docker I get the error mentioned in my first post.

However, after trying it again today it seems that authentication worked inside docker as well. I didn't do anything different though, this is quite strange.
Could it be that the server was somehow blocking my authentication attempts? Maybe because on my host I authenticated to quite a few conan remotes? 

Thanks!

**Comment by memsharded** [comment]
The really weird thing is the response:
```
ERROR: b'<html>\r\n<head><title>403 Forbidden</title></head>\r\n<body>\r\n<center><h1>403 Forbidden</h1></center>\r\n</body>\r\n</html>\r\n'
```
this is html, which I think the Conan server API will never respond. That means there could be someone else in the middle, like any other server, nginx, redirect, or something that is sending the Conan requests to a different place. You might try to use another random remote URL instead of your correct one, and see what response you get.

**Comment by andreear-23** [comment]
Yes, I think there might be some in-between server that deals with the the authentication, it's SSO based. 
What other random remote URL do you mean? Another one from the private artifactory or?

**Comment by memsharded** [comment]
> What other random remote URL do you mean? Another one from the private artifactory or?

Yes, whatever. Just invent one like ``https://<private_domain>.com/artifactorynonexist/....`` and see what happens.

**Comment by andreear-23** [comment]
Ok yeah, so that gives the same error. I think there might be something wrong on the server-side configuration. I just tried to re-login now and it gives the same error for both the correct URL and the wrong one. 
Then I tried to use a different mirror of the artifactory and that one worked. 
I just wasn't sure if there was maybe some additional parameters I should have started the docker container with. I think all is clear now though, we can close this issue :).

Thanks a lot for all your help!
